Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp

```diff
@@ -124,20 +125,16 @@ std::optional<CertificateSummary> CertificateInfo::summary() const
     if (subjectName)
         summaryInfo.subject = String::fromUTF8(subjectName.get());
     if (dnsNames) {
-        for (unsigned i = 0; i < dnsNames->len; ++i) {
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port
-            GBytes* bytes = static_cast<GBytes*>(dnsNames->pdata[i]);
-            WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+        for (auto dnsName : span(dnsNames)) {
+            GBytes* bytes = static_cast<GBytes*>(dnsName);
             gsize dataLength;
             const auto* data = g_bytes_get_data(bytes, &dataLength);
             summaryInfo.dnsNames.append(String({ static_cast<const char*>(data), dataLength }));
         }
```

Comment:
Then here we can pass the type of the pointer array elements and remove the manual cast. Also, as a reminder we have a `span(GBytes*)` helper, which brings in further simplification:

```suggestion
        for (auto dnsName* : span<GBytes*>(dnsNames))
            summaryInfo.dnsNames.append(String(span(dnsName)));
```